File: 1350-remove-sub-folders-from-the-filesystem/remove-sub-folders-from-the-filesystem.ts
function removeSubfolders(folder: string[]): string[] {
    const set = new Set(folder);
    const array = Array.from(set);
    array.sort()
    let i = 0, j = 1;
    while(j < array.length) {
        if(array[j].startsWith(array[i] + '/')) {
            set.delete(array[j])
            j++
        } else {
            i = j++
        }
    }
    return Array.from(set)
};

File: 1350-remove-sub-folders-from-the-filesystem/remove-sub-folders-from-the-filesystem.rs
impl Solution {
    pub fn remove_subfolders(mut folder: Vec<String>) -> Vec<String> {
        folder.sort();
        let mut root_str = "*".to_string();
        let mut result = vec![];
        for (i, v) in folder.iter().enumerate() {
            if !v.starts_with(&root_str) {
                result.push(v.to_string());
                root_str = format!("{}/", v);
            }
        }
        result
    }
}